pkg/kubelet/cm/dra/plugin/client_test.go

```diff
@@ -0,0 +1,140 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc"
+	drapbv1 "k8s.io/kubelet/pkg/apis/dra/v1alpha2"
+)
+
+type fakeGRPCServer struct {
+	drapbv1.UnimplementedNodeServer
+}
+
+func (f *fakeGRPCServer) NodePrepareResource(ctx context.Context, in *drapbv1.NodePrepareResourceRequest) (*drapbv1.NodePrepareResourceResponse, error) {
+	return &drapbv1.NodePrepareResourceResponse{CdiDevices: []string{"dummy"}}, nil
+}
+
+func (f *fakeGRPCServer) NodeUnprepareResource(ctx context.Context, in *drapbv1.NodeUnprepareResourceRequest) (*drapbv1.NodeUnprepareResourceResponse, error) {
+	return &drapbv1.NodeUnprepareResourceResponse{}, nil
+}
+
+type tearDown func()
+
+func setupFakeGRPCServer(t *testing.T) (string, tearDown) {
+	p, err := os.MkdirTemp("", "dra_plugin")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	closeCh := make(chan struct{})
+	addr := filepath.Join(p, "server.sock")
+	teardown := func() {
+		close(closeCh)
+		os.RemoveAll(addr)
+	}
+
+	listener, err := net.Listen("unix", addr)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	s := grpc.NewServer()
+	fakeGRPCServer := &fakeGRPCServer{}
+	drapbv1.RegisterNodeServer(s, fakeGRPCServer)
+
+	go func() {
+		go s.Serve(listener)
+		<-closeCh
+		s.GracefulStop()
+	}()
+
+	return addr, teardown
+}
+
+func TestGRPCConnIsReused(t *testing.T) {
+	addr, teardown := setupFakeGRPCServer(t)
+	witness := make(map[string]int)
+	wg := sync.WaitGroup{}
+	m := sync.Mutex{}
+
+	plugin := &Plugin{
+		endpoint: addr,
+	}
+
+	if err := plugin.CreateGRPCConn(); err != nil {
+		t.Fatal(err)
+	}
+
+	ctime := time.After(100 * time.Millisecond)
+	// wait a bit for the connection to be in ready state before performing our test
+	for !plugin.IsConnReady() {
+		select {
+		case <-ctime:
+			t.Fatal("gRPC connection was not ready")
+		default:
+		}
+		time.Sleep(time.Millisecond)
+	}
+
+	// ensure the plugin we are using is registered
+	draPlugins.Set("dummy-plugin", plugin)
+
+	// we call `NodePrepareResource` 100 times and check whether a new connection is created or the same is reused
+	for i := 0; i < 100; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			client, err := NewDRAPluginClient("dummy-plugin")
+			if err != nil {
+				t.Fail()
+				return
+			}
+			client.NodePrepareResource(context.TODO(), "dummy-namespace", "dummy-uid", "dummy-claim", "dummy-resource")
+
+			client.(*draPluginClient).plugin.Lock()
+			p := fmt.Sprintf("%p", client.(*draPluginClient).plugin.conn)
```

Comment:
Instead of formatting this into a string, why not just define the map as:
```
reusedConns := make(map[*grpc.ClientConn]int)
```
It would make it clearer what it is used for anyway.

Reply:
done in ef1a07b143e